lib/metadata/bucketfile/backend.js

```diff
@@ -0,0 +1,302 @@
+            });
+            return undefined;
+        });
+    }
+
+    putBucketAttributes(bucketName, bucketMD, log, cb) {
+        this.loadMetastore(err => {
+            if (err) {
+                return cb(err);
+            }
+            this.metastore.put(bucketName, bucketMD.serialize(), OPTIONS, cb);
+            return undefined;
+        });
+    }
+
+    deleteBucket(bucketName, log, cb) {
+        this.loadMetastore(err => {
+            if (err) {
+                return cb(err);
+            }
+            return cb(null);
+        });
+    }
+
+    putObject(bucketName, objName, objVal, log, cb) {
+        this.loadDBIfExists(bucketName, log, err => {
+            if (err) {
+                return cb(err);
+            }
+            this.dbs[bucketName].put(objName, JSON.stringify(objVal),
+                                     OPTIONS, cb);
+            return undefined;
+        });
+    }
+
+    getObject(bucketName, objName, log, cb) {
+        this.loadDBIfExists(bucketName, log, err => {
+            if (err) {
+                return cb(err);
+            }
+            this.dbs[bucketName].get(objName, (err, data) => {
+                if (err && err.notFound) {
+                    return cb(errors.NoSuchObject);
+                }
+                return cb(null, JSON.parse(data));
+            });
+            return undefined;
+        });
+    }
+
+    deleteObject(bucketName, objName, log, cb) {
+        this.loadDBIfExists(bucketName, log, err => {
+            if (err) {
+                return cb(err);
+            }
+            this.dbs[bucketName].del(objName, OPTIONS, cb);
+            return undefined;
+        });
+    }
+
+    /**
+     *  This function checks if params have a property name
+     *  If there is add it to the finalParams
+     *  Else do nothing
+     *  @param {String} name - The parameter name
+     *  @param {Object} params - The params to search
+     *  @param {Object} extParams - The params sent to the extension
+     *  @return {undefined}
+     */
+    addExtensionParam(name, params, extParams) {
+        if (params.hasOwnProperty(name)) {
+            extParams[name] = params[name];
+        }
+    }
+
+    /**
+     *  This complex function deals with different extensions of bucket listing:
+     *  Delimiter based search or MPU based search.
+     *  @param {String} bucketName - The name of the bucket to list
+     *  @param {Object} params - The params to search
+     *  @param {Object} log - The logger object
+     *  @param {function} cb - Callback when done
+     *  @return {undefined}
+     */
+    internalListObject(bucketName, params, log, cb) {
+        const requestParams = {};
+        let Ext;
+        let extParams = {};
+        // multipart upload listing
+        if (params.listingType === 'multipartuploads') {
+            Ext = arsenal.listMPU.ListMultipartUploads;
+            this.addExtensionParam('queryPrefixLength', params, extParams);
```

Comment:
This code comes from MetaData and there is a bit of context here. I believe extParams has a list of params supported by levelDB and we want to assign/change only those parameters based on the request. Isolating that logic into its own method(even though its just one line) would be more meaningful as we can expand on it later.